ethereum/contracts/IncentivizedOutboundChannel.sol

```diff
@@ -2,25 +2,85 @@
 pragma solidity >=0.7.6;
 pragma experimental ABIEncoderV2;
 
+import "@openzeppelin/contracts/access/AccessControl.sol";
 import "./OutboundChannel.sol";
+import "./ChannelAccess.sol";
+import "./FeeSource.sol";
 
 // IncentivizedOutboundChannel is a channel that sends ordered messages with an increasing nonce. It will have incentivization too.
-contract IncentivizedOutboundChannel is OutboundChannel {
+contract IncentivizedOutboundChannel is OutboundChannel, ChannelAccess, AccessControl {
+
+    // Governance contracts will administer using this role.
+    bytes32 public constant CONFIG_UPDATE_ROLE = keccak256("CONFIG_UPDATE_ROLE");
 
     // Nonce for last submitted message
     uint64 public nonce;
 
+    uint256 private fee;
+    FeeSource private feeSource;
+
     event Message(
         address source,
         uint64  nonce,
+        uint256 fee,
         bytes   payload
     );
 
+    event FeeChanged(
+        uint256 oldFee,
+        uint256 newFee
+    );
+
+    constructor() {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    // Once-off post-construction call to set initial configuration.
+    function initialize(
+        address _configUpdater,
+        address _feeSource,
+        address[] memory defaultOperators
+    )
+    external {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), "Caller is unauthorized");
+
+        // Set initial configuration
+        feeSource = FeeSource(_feeSource);
+        grantRole(CONFIG_UPDATE_ROLE, _configUpdater);
+        for (uint i = 0; i < defaultOperators.length; i++) {
+            _authorizeDefaultOperator(operator);
+        }
+
+        // drop admin privileges
+        renounceRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    // Update message submission fee.
+    function setFee(uint256 _amount) external {
+        require(hasRole(CONFIG_UPDATE_ROLE, msg.sender), "Caller is unauthorized");
```

Comment:
can use a modifier for this like onlyConfigUpdater - https://docs.soliditylang.org/en/v0.6.2/contracts.html#function-modifiers

Reply:
Will keep it mind when I flesh out governance on the Ethereum side.